error with truncate partition

```
Caused by: java.io.IOException: Error processing file://iris.parquet/
        at com.altinity.ice.cli.internal.cmd.Insert.lambda$run$2(Insert.java:215)
 [4 skipped]
        ... 1 common frames omitted
Caused by: java.lang.ClassCastException: class org.apache.parquet.io.api.Binary$ByteArrayBackedBinary cannot be cast to class java.lang.CharSequence (org.apache.parquet.io.api.Binary$ByteArrayBackedBinary is in unnamed module of loader org.codehaus.mojo.exec.URLClassLoaderBuilder$ExecJavaClassLoader @4091b9c3; java.lang.CharSequence is in module java.base of loader 'bootstrap')
        at org.apache.iceberg.transforms.Truncate$TruncateString.apply(Truncate.java:281)
        at com.altinity.ice.cli.internal.iceberg.Partitioning.inferPartitionKey(Partitioning.java:182)
        at com.altinity.ice.cli.internal.cmd.Insert.processFile(Insert.java:426)
        at com.altinity.ice.cli.internal.cmd.Insert.lambda$run$2(Insert.java:194)
        ... 4 common frames omitted
```

--- a/ice-rest-catalog/src/test/resources/scenarios/insert-partitioned/run.sh.tmpl
+++ b/ice-rest-catalog/src/test/resources/scenarios/insert-partitioned/run.sh.tmpl
@@ -85,6 +85,33 @@ if [[ "${FILES_OUT_BUCKET}" != *${EXPECTED_DATA_PATH_BUCKET}* ]]; then
 fi
 echo "OK Validated correct bucket-partitioned data file path for ${TABLE_NAME_BUCKET}"
 
+# Create table with truncate partition and insert data
+{{ICE_CLI}} --config {{CLI_CONFIG}} insert --create-table ${TABLE_NAME_TRUNCATE} ${INPUT_PATH_IRIS} --partition="${PARTITION_SPEC_TRUNCATE}"
+echo "OK Inserted data with truncate partitioning into table ${TABLE_NAME_TRUNCATE}"
+
+# List partitions and validate output
+LIST_PARTITIONS_OUT_TRUNCATE=$({{ICE_CLI}} --config {{CLI_CONFIG}} list-partitions "${TABLE_NAME_TRUNCATE}")
+if ! grep -q "partitions:" <<<"${LIST_PARTITIONS_OUT_TRUNCATE}"; then
+  echo "FAIL: list-partitions output missing 'partitions:' section"
+  printf '%s\n' "${LIST_PARTITIONS_OUT_TRUNCATE}"
+  exit 1
+fi
+if ! grep -q "variety_trunc=" <<<"${LIST_PARTITIONS_OUT_TRUNCATE}"; then
+  echo "FAIL: list-partitions output missing truncate partition entries (expected 'variety_trunc=')"
+  printf '%s\n' "${LIST_PARTITIONS_OUT_TRUNCATE}"
+  exit 1
+fi
+echo "OK Listed and validated truncate partitions for ${TABLE_NAME_TRUNCATE}"
+
+# Validate data file was inserted to correct truncate-partitioned path
+FILES_OUT_TRUNCATE=$({{ICE_CLI}} --config {{CLI_CONFIG}} files ${TABLE_NAME_TRUNCATE})
+
+if [[ "${FILES_OUT_TRUNCATE}" != *${EXPECTED_DATA_PATH_TRUNCATE}* ]]; then
+  echo "FAIL: expected substring '${EXPECTED_DATA_PATH_TRUNCATE}' not found in files command output: ${FILES_OUT_TRUNCATE}"
+  exit 1
+fi
+echo "OK Validated correct truncate-partitioned data file path for ${TABLE_NAME_TRUNCATE}"
+
 # Cleanup
 {{ICE_CLI}} --config {{CLI_CONFIG}} delete-table ${TABLE_NAME_IRIS}
 echo "OK Deleted table: ${TABLE_NAME_IRIS}"
@@ -92,6 +119,8 @@ echo "OK Deleted table: ${TABLE_NAME_IRIS}"
 echo "OK Deleted table: ${TABLE_NAME_ONTIME}"
 {{ICE_CLI}} --config {{CLI_CONFIG}} delete-table ${TABLE_NAME_BUCKET}
 echo "OK Deleted table: ${TABLE_NAME_BUCKET}"
+{{ICE_CLI}} --config {{CLI_CONFIG}} delete-table ${TABLE_NAME_TRUNCATE}
+echo "OK Deleted table: ${TABLE_NAME_TRUNCATE}"
 
 {{ICE_CLI}} --config {{CLI_CONFIG}} delete-namespace ${NAMESPACE_NAME}
 echo "OK Deleted namespace: ${NAMESPACE_NAME}"

--- a/ice-rest-catalog/src/test/resources/scenarios/insert-partitioned/scenario.yaml
+++ b/ice-rest-catalog/src/test/resources/scenarios/insert-partitioned/scenario.yaml
@@ -17,4 +17,6 @@ env:
   PARTITION_SPEC_BUCKET: '[{"column":"variety","transform":"bucket[3]"}]'
   # Scheme-agnostic: files output may use s3:// or s3a:// depending on Hadoop FS
   EXPECTED_DATA_PATH_BUCKET: "iris_bucket_partitioned/data/variety_bucket="
-
+  TABLE_NAME_TRUNCATE: "test_insert_partitioned.iris_truncate_partitioned"
+  PARTITION_SPEC_TRUNCATE: '[{"column":"variety","transform":"truncate[2]"}]'
+  EXPECTED_DATA_PATH_TRUNCATE: "iris_truncate_partitioned/data/variety_trunc="

--- a/ice/src/main/java/com/altinity/ice/cli/internal/iceberg/Partitioning.java
+++ b/ice/src/main/java/com/altinity/ice/cli/internal/iceberg/Partitioning.java
@@ -353,7 +353,7 @@ public final class Partitioning {
                   sourceFieldName, toGenericRecordFieldValue(value, fieldSpec.type()));
               break;
             default:
-              if (transformName.startsWith("bucket[")) {
+              if (transformName.startsWith("bucket[") || transformName.startsWith("truncate[")) {
                 partitionRecord.setField(
                     sourceFieldName, toGenericRecordFieldValue(value, fieldSpec.type()));
               } else {

